assets/js/blocks/rating-filter/block.tsx

```diff
@@ -189,12 +190,16 @@ const RatingFilterBlock = ( {
 		if ( filteredCountsLoading || blockAttributes.isPreview ) {
 			return;
 		}
+
+		const defaultPreviewCounts = isEditor ? editorPreviewOptions : [];
+
 		const orderedRatings =
 			! filteredCountsLoading &&
 			objectHasProp( filteredCounts, 'rating_counts' ) &&
-			Array.isArray( filteredCounts.rating_counts )
+			Array.isArray( filteredCounts.rating_counts ) &&
+			filteredCounts.rating_counts.length > 0
 				? [ ...filteredCounts.rating_counts ].reverse()
-				: [];
```

Comment:
I commented on the `preview.tsx` file but want to explain it more here, I think we can reuse the `previewOptions` here by setting the `displayedOptions` to `previewOptions` if the `orderedRatings` is empty (inside the Editor only). We will need another state to track if the store has rating or not to display the notice.

I think by doing so, we can reuse more code and make this PR smaller.

Reply:
Makes sense: as mentioned in [my response](https://github.com/woocommerce/woocommerce-blocks/pull/7763#discussion_r1045549717) on the `preview.tsx` file, I initially kept them separated in case we wanted to display different content on preview mode x the editor in the future, but since at this point they don't differ, I've adopted your suggestion here and consolidated both